analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -0,0 +1,93 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Helpers;
+
+public class SonarLintXmlReader
+{
+    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+
+    private readonly SonarLintXml sonarLintXml;
+    private readonly string propertyLanguage;
+
+    private bool? ignoreHeaderComments;
+    public bool IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+
+    private bool? analyzeGeneratedCode;
+    public bool AnalyzeGeneratedCode => analyzeGeneratedCode ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.analyzeGeneratedCode"));
+
+    private string[] exclusions;
+    public string[] Exclusions => exclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.exclusions"));
+
+    private string[] inclusions;
+    public string[] Inclusions => inclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.inclusions"));
+
+    private string[] globalExclusions;
+    public string[] GlobalExclusions => globalExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.global.exclusions"));
+
+    private string[] testExclusions;
+    public string[] TestExclusions => testExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.test.exclusions"));
+
+    private string[] testInclusions;
+    public string[] TestInclusions => testInclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.test.inclusions"));
+
+    private string[] globalTestExclusions;
+    public string[] GlobalTestExclusions => globalTestExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.global.test.exclusions"));
+
+    public SonarLintXmlReader(SourceText sonarLintXml, string language)
+    {
+        this.sonarLintXml = sonarLintXml == null ? SonarLintXml.Empty : ParseContent(sonarLintXml);
+        propertyLanguage = language == LanguageNames.CSharp ? "cs" : "vbnet";
+    }
+
+    private static SonarLintXml ParseContent(SourceText sonarLintXml)
+    {
+        try
+        {
+            var serializer = new XmlSerializer(typeof(SonarLintXml));
+            var byteArray = Encoding.UTF8.GetBytes(sonarLintXml.ToString());
+            var stream = new MemoryStream(byteArray);
+            using var sr = new StreamReader(stream, Encoding.UTF8, false);
+            using var reader = XmlReader.Create(sr);
+            return (SonarLintXml)serializer.Deserialize(reader);
+        }
+        catch
+        {
+            return SonarLintXml.Empty;
+        }
+    }
+
+    private string ReadProperty(string property) =>
+        sonarLintXml is { Settings: { } settings }
+        ? settings.Where(x => x.Key.Equals(property)).Select(x => x.Value).FirstOrDefault()
+        : string.Empty;
```

Comment:
Not sure why this would return "" when the 1st branch can return `null`
```suggestion
        : null;
```

Reply:
https://github.com/SonarSource/sonar-dotnet/pull/6915

Reply:
I don't see this in #6915

Reply:
🤦  fixed - probably lost in refactoring.